bin/crystal

```diff
@@ -94,27 +94,22 @@ _canonicalize_file_path() {
 
 ##############################################################################
 
-# Based on http://stackoverflow.com/a/2108540/2199687
+# Based on http://stackoverflow.com/q/370047/641451
 remove_path_item() {
-  local path item work
+  local path item
 
   path="$1"
   item="$2"
 
-  work=":$path:"
-  work="${work/:$item:/:}"
-  work="${work%:}"
-  work="${work#:}"
-
-  echo "$work"
+  echo -n $path | awk -v RS=: -v ORS=: '$0 != "'$item'"' | sed 's/:$//'
 }
 
 ##############################################################################
 
 __has_colors() {
   local num_colors=$(tput colors 2>/dev/null)
 
-  if [ "$num_colors" -gt 2 ]; then
+  if [ "0$num_colors" -gt 2 ]; then
```

Comment:
Oh I see the trick! I think it's better to avoid tricks, readability should be king everywhere.

To check for empty string, you can use `[ -z "$num_colors" ]` 😃